glass/shapes.py

```diff
@@ -142,17 +142,24 @@ def ellipticity_ryden04(mu, sigma, gamma, sigma_gamma, size=None, *, rng=None):
     if size is None:
         size = np.broadcast(mu, sigma, gamma, sigma_gamma).shape
 
+    # broadcast all inputs to output shape
+    # this makes it possible to efficiently resample later
+    mu = np.broadcast_to(mu, size, subok=True)
+    sigma = np.broadcast_to(sigma, size, subok=True)
+    gamma = np.broadcast_to(gamma, size, subok=True)
+    sigma_gamma = np.broadcast_to(sigma_gamma, size, subok=True)
+
     # draw gamma and epsilon from truncated normal -- eq.s (10)-(11)
     # first sample unbounded normal, then rejection sample truncation
     eps = rng.normal(mu, sigma, size=size)
     bad = eps > 0
     while np.any(bad):
-        eps[bad] = rng.normal(mu, sigma, size=size)[bad]
+        eps[bad] = rng.normal(mu[bad], sigma[bad])
         bad = eps > 0
```

Comment:
While we are at it, this might be one of the few times where the walrus is appropriate?

```py
    while np.any(bad := eps > 0):
        eps[bad] = rng.normal(mu[bad], sigma[bad])
```